integration/didweb_integration_test.go

```diff
@@ -0,0 +1,153 @@
+package integration
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+var didWebContext = NewTestContext("DidWeb")
```

Comment:
interesting...I'm hesitant of having global state like this. there could be side effects between tests which could produce some ugly errors.

what do you think about instantiating a new context per test? also nit `DIDWeb`

Reply:
I need to have shared state for the tests as some inputs are needed from previous tests.

I added a mutex around this and since each test is in it's own context there should be no interference betwen tests